x/concentrated-liquidity/lp.go

```diff
@@ -75,6 +75,11 @@ func (k Keeper) createPosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddr
 		if err := k.initializeFeeAccumulatorPosition(cacheCtx, poolId, owner, lowerTick, upperTick); err != nil {
 			return sdk.Int{}, sdk.Int{}, sdk.Dec{}, err
 		}
+
+		err = k.initPositionUptime(ctx, poolId, owner, lowerTick, upperTick, liquidityDelta, joinTime, freezeDuration)
+		if err != nil {
+			return sdk.Int{}, sdk.Int{}, sdk.Dec{}, err
+		}
```

Comment:
It seems that the semantics of position being "initial" is different between incentives and fees.

As a result, I don't think we should be adding the call to `initPositionUptime` here.

Based on the description of the issue, what I think should happen is the following:
- outside of this position length branch, check if `hasFullPosition` returns true. 
    - If yes, check if it is still frozen.
       * if yes, return error
       * if not, continue by adding to it
    - if not, call `initPositionUptime`

Reply:
I thought the point of `initPositionUptime` here was that we expect there to be no full position initialized in that specific tick range with the owner. It it does then we create a new one with different joinTime. 

In what cases would `hasFullPosition` if there is no `InitPositionUptime` initialized?

Reply:
ooh i see, its cause `getAllPositionsWithVaryingFreezeTimes` doesnot return FullPosition it just returns position based on lower, upper ticks, owner, poolId. So, we might run into a case where at that joinTime is not initialized but it is initialized at that tick range